### PR TITLE
 Using golang protobuf library

### PR DESCRIPTION
**Pull request by knabben**
Fixes https://github.com/kubernetes-sigs/kpng/issues/149

`server/pkg/proxystore/proxystore.go` and `watchstate` are using gogo library (https://github.com/gogo/protobuf) to parse the protobuf, this version is not fully compatible with the current binary build with (https://pkg.go.dev/google.golang.org/protobuf/proto), this PR changes both files to use `google protobuf` instead.

### CODE REVIEW

File: server/pkg/server/watchstate/watchstate.go [hunk 1]
```diff
@@ -101,16 +99,16 @@ func (w *WatchState) send(item *localnetv1.OpItem) {
 }
 
 func (w *WatchState) sendSet(set localnetv1.Set, path string, m proto.Message) {
-	w.pb.Reset()
```
Comment (jayunit100) [inline server/pkg/server/watchstate/watchstate.go:104]: wait , no more reset ? whys that

Comment (knabben) [reply server/pkg/server/watchstate/watchstate.go:104]: adding a reset at this point will clear new requests..

Comment (knabben) [reply server/pkg/server/watchstate/watchstate.go:104]: this reset in particular is from the buffer and not the message..
adding a `proto.Reset(m)` at this point will clear new request so they come up as empty.

File: server/pkg/proxystore/proxystore.go [hunk 2]
```diff
@@ -61,16 +59,16 @@ func New() *Store {
...
 func (s *Store) hashOf(m proto.Message) (h uint64) {
-	if err := s.pb.Marshal(m); err != nil {
+	message, err := proto.Marshal(m)
+	if err != nil {
 		panic(err) // should not happen
 	}
-	h = xxhash.Sum64(s.pb.Bytes())
-	s.pb.Reset()
+	h = xxhash.Sum64(message)
+	proto.Reset(m)
```
Comment (mcluseau) [inline server/pkg/proxystore/proxystore.go:71]: the original `Reset` was resetting the *buffer*, we don't want to reset the given message ;-) With `proto.Marshal` you have a new []byte every time, so no need to reset anything.

File: server/pkg/server/watchstate/watchstate.go [hunk 3]
```diff
@@ -31,8 +31,7 @@ type WatchState struct {
 	res   localnetv1.OpSink
 	sets  []localnetv1.Set
 	diffs []*diffstore.DiffStore
-	pb    *proto.Buffer
-	Err   error
+	Err error
```
Comment (mcluseau) [inline server/pkg/server/watchstate/watchstate.go:34]: I think there's a minor go fmt issue here, `error` should be aligned.

**Review by mcluseau — CHANGES_REQUESTED** [review]

**Pull request merged**